connector/src/main/java/org/jboss/as/connector/services/resourceadapters/DirectConnectionFactoryActivatorService.java

```diff
@@ -186,13 +182,13 @@ public void start(org.jboss.msc.service.StartContext context) throws org.jboss.m
             Security security = null;
             if (securitySetting != null) {
                 if ("".equals(securitySetting)) {
-                    security = new SecurityImpl(null, null, false, false);
+                    SUBSYSTEM_RA_LOGGER.legacySecurityNotSupported();
                 } else if ("application".equals(securitySetting)) {
-                    security = new SecurityImpl(null, null, true, false);
+                    SUBSYSTEM_RA_LOGGER.legacySecurityNotSupported();
```

Comment:
This method produces an OperationFailedException but it is not thrown. I don't think throwing OFE from here is correct; some other exception type is needed.